scripts/thief.bicep

```diff
@@ -0,0 +1,346 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for
+// full license information.
+
+param location string
+param prefix string
+param thief_runs_resource_group string
+param thief_app_insights_connection_string string
+param thief_container_registry_host string
+param thief_container_registry_password string
+param thief_container_registry_shortname string
+param thief_container_registry_user string
+param user_principal_id string
+param thief_shared_subscription_id string
+param thief_shared_keyvault_name string
```

Comment:
Would be good to add min and max length for KV resource
Ex:

param KeyVaultName string {
  default: '${resourceGroup().name}-kv'
  minLength: 3
  maxLength: 24
  metadata: {
    description: 'The name of the key vault for xxx'
  }
}

Reply:
Agree that this could be useful, but I don't want to introduce any limits at this point.

Reply:
The good reason to do it is it'll fail immediately if the parameters aren't right. These are known limits in Azure for KV.

If you don't check it, and someone passes a bad value, it'll take longer because it must deploy and fail. If it fails, you may have some resources done, but not others, and that requires clean up.

Reply:
OK, you talked me into it.